src/tools/crawler/main.rs

```diff
@@ -108,15 +109,18 @@ async fn main() {
                 }
             }
 
-            crawler.send_broadcast(Message::GetAddr).unwrap();
+            crawler.send_broadcast(Message::GetAddr).expect("writing hasn't been enabled");
 
             if crawler.known_network.num_connections() > 0 {
                 // Update graph, then create a summary and log it to a file.
                 network_metrics.update_graph(&crawler);
                 let network_summary = network_metrics.request_summary(&crawler);
 
                 info!("{}", network_summary);
-                network_summary.log_to_file().unwrap();
+                network_summary.log_to_file().unwrap_or_else(|e| {
+                    error!(parent: crawler.node().span(), "Couldn't write summary to file: {}", e);
+                    ()
+                });
```

Comment:
this would be expressed better as `if let Err(e) = network_summary.log_to_file() { error!(...) }`